### PR TITLE
sci-astronomy/stellarium: add 0.22.1

### PR DESCRIPTION
**Pull request by Kangie**
Reapplied existing patches due to failure and saved a new diff.

As `dev-qt/qtcharts` is a new dependency (and is untested on ppc), ppc is now masked and all other arches are marked as testing.
 
Signed-off-by: Matt Jolly <Matt.Jolly@footclan.ninja>

### CODE REVIEW

File: sci-astronomy/stellarium/stellarium-0.22.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} )
+inherit cmake desktop flag-o-matic python-any-r1 xdg virtualx
+
+DESCRIPTION="3D photo-realistic skies in real time"
+HOMEPAGE="https://stellarium.org/"
+MY_DSO_VERSION="3.13"
```
Comment (DarthGandalf) [inline sci-astronomy/stellarium/stellarium-0.22.1.ebuild:11]: Have you tried building it (and running) with USE=deepsky?

Stellarium 0.22.1 uses DSO 3.15: https://github.com/Stellarium/stellarium-data/releases/tag/dso-3.15

And it warns about the wrong DSO version during startup

**Comment by Kangie** [comment]
Bug: https://bugs.gentoo.org/838916

For rekeywording on ppc

**Review by DarthGandalf — COMMENTED** [review]
Thanks!

File: sci-astronomy/stellarium/stellarium-0.22.1.ebuild (hunk 1 above)
Comment (Kangie) [reply sci-astronomy/stellarium/stellarium-0.22.1.ebuild:11]: I did not, but I'll give it a go now and report back.

Comment (Kangie) [reply sci-astronomy/stellarium/stellarium-0.22.1.ebuild]: Good catch; it was throwing a warning. Simple fix in the ebuild though!

**Comment by Kangie** [comment]
Hm. This segfaults if 'stars' is enabled. I'll do some digging.

**Comment by DarthGandalf** [comment]
> Hm. This segfaults if 'stars' is enabled. I'll do some digging.

It works for me...

**Comment by Kangie** [comment]
> > Hm. This segfaults if 'stars' is enabled. I'll do some digging.
> 
> It works for me...

Yeah, Seems to have been some weirdness with adding the use flag when it didn't already exist maybe?

I've been unable to replicate again, but it's not a long compile so I might delete all the configs and data and see what happens if the flag is enabled and then rebuild *after* the app has done it's first run init.

**Review by DarthGandalf — APPROVED** [review]

**Comment by Kangie** [comment]
Can confirm - if built with `USE=-stars` and then built with `USE=stars`, on first launch stellarium will segfault, immediately after importing the stars DBs.

```
. . .
Loading nomenclature for Solar system bodies ...
Loaded 15781 / 15783 items of planetary surface nomenclature
WARNING - These planets to assign nomenclature items were not found: "Dactyl"
Loading star data ...
"Loading /usr/share/stellarium/stars/default/stars_0_0v0_8.cat: 0_0v0_8; 4979"
"Loading /usr/share/stellarium/stars/default/stars_1_0v0_8.cat: 1_0v0_8; 21806"
"Loading /usr/share/stellarium/stars/default/stars_2_0v0_8.cat: 2_0v0_8; 150826"
"Loading /usr/share/stellarium/stars/default/stars_3_1v0_4.cat: 3_1v0_4; 425807"
Found file  "/usr/share/stellarium/stars/default/stars_4_1v0_2.cat" , checking md5sum..
MD5 sum correct!
"Loading /usr/share/stellarium/stars/default/stars_4_1v0_2.cat: 4_1v0_4; 1692779"
Found file  "/usr/share/stellarium/stars/default/stars_5_2v0_1.cat" , checking md5sum..
MD5 sum correct!
[1]    1741584 segmentation fault (core dumped)  stellarium --verbose
```

Everything seems to work fine from there - if you launch the app again it just works; seems like a problem for upstream.

**Comment by DarthGandalf** [comment]
Confirmed. Except that it also crashes after disabling stars.

```
Loaded 103 Solar System bodies                                                                                                                                                                                                                                                    
File ssystem_minor.ini is loaded successfully...                                                                                                                                                                                                                                  
Loading nomenclature for Solar system bodies ...                                                                                                                                                                                                                                  
Loaded 15732 / 15783 items of planetary surface nomenclature                                                                                                                                                                                                                      
WARNING - These planets to assign nomenclature items were not found: "Bennu, Dactyl, Ryugu"                                                                                                                                                                                       
Loading star data ...                                                                                                                                                                                                                                                             
"Loading /usr/share/stellarium/stars/default/stars_0_0v0_8.cat: 0_0v0_8; 4979"                                                                                                                                                                                                    
"Loading /usr/share/stellarium/stars/default/stars_1_0v0_8.cat: 1_0v0_8; 21806"                                                                                                                                                                                                   
"Loading /usr/share/stellarium/stars/default/stars_2_0v0_8.cat: 2_0v0_8; 150826"                                                                                                                                                                                                  
"Loading /usr/share/stellarium/stars/default/stars_3_1v0_4.cat: 3_1v0_4; 425807"                                                                                                                                                                                                  
"Warning: could not find star catalog stars/default/stars_4_1v0_2.cat"                                                                                                                                                                                                            
"Warning: could not find star catalog stars/default/stars_5_2v0_1.cat"                                                                                                                                                                                                            
[1]    29601 segmentation fault  stellarium
```

**Comment by DarthGandalf** [comment]
<details>
<summary>gdb log when enabling stars</summary>

```
#0  QVariant::toMap (this=0x7400610063002e) at /mnt/portagetmp/portage/dev-qt/qtcore-5.15.2-r14/work/qtbase-f4ac0b55c37f2b594ffbe639db43dac365825c7c/src/corelib/kernel/qvariant.cpp:2721
No locals.
#1  0x0000555555aae68a in StarMgr::setCheckFlag (this=0x5555583d8220, catId=..., b=true) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/core/modules/StarMgr.cpp:587
        m = {d = 0x5555587a8f90}
        checked = <optimized out>
        tmp = {<QFileDevice> = {<QIODevice> = {<QObject> = {_vptr.QObject = 0x7ffff6a9bf00 <vtable for QObject+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, 
                    data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x5555587ad860}, 
                static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                  superdata = {direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff6964720 <qt_meta_stringdata_QIODevice>, data = 0x7ffff6964620 <qt_meta_data_QIODevice>, 
                  static_metacall = 0x7ffff6717010 <QIODevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a98980 <QIODevice::staticMetaObject>}, 
                stringdata = 0x7ffff69645a0 <qt_meta_stringdata_QFileDevice>, data = 0x7ffff6964560 <qt_meta_data_QFileDevice>, static_metacall = 0x7ffff6711340 <QFileDevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, 
                extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a987e0 <QFileDevice::staticMetaObject>}, stringdata = 0x7ffff69643e0 <qt_meta_stringdata_QFile>, data = 0x7ffff69643a0 <qt_meta_data_QFile>, 
              static_metacall = 0x7ffff670dfe0 <QFile::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}
        catV = <optimized out>
        __for_range = <optimized out>
        __for_begin = <optimized out>
        __for_end = <optimized out>
        idx = 7
#2  0x0000555555ab504a in StarMgr::checkAndLoadCatalog (this=0x5555583d8220, catDesc=...) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/core/modules/StarMgr.cpp:559
        md5Hash = {d = 0x555556b66c10}
        cat_sz = <optimized out>
        maxStarBufMd5 = <optimized out>
        cat = <optimized out>
        file = {<QFileDevice> = {<QIODevice> = {<QObject> = {_vptr.QObject = 0x7ffff6a98518 <vtable for QFile+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, 
                    data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x5555581d5c00}, 
                static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                  superdata = {direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff6964720 <qt_meta_stringdata_QIODevice>, data = 0x7ffff6964620 <qt_meta_data_QIODevice>, 
                  static_metacall = 0x7ffff6717010 <QIODevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a98980 <QIODevice::staticMetaObject>}, 
                stringdata = 0x7ffff69645a0 <qt_meta_stringdata_QFileDevice>, data = 0x7ffff6964560 <qt_meta_data_QFileDevice>, static_metacall = 0x7ffff6711340 <QFileDevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, 
                extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a987e0 <QFileDevice::staticMetaObject>}, stringdata = 0x7ffff69643e0 <qt_meta_stringdata_QFile>, data = 0x7ffff69643a0 <qt_meta_data_QFile>, 
              static_metacall = 0x7ffff670dfe0 <QFile::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}
        checked = false
        catalogFileName = {d = 0x5555587a7840}
        catalogFilePath = {d = 0x5555587ad750}
        z = <optimized out>
#3  0x0000555555ab5702 in StarMgr::loadData (this=0x5555583d8220, starsConfig=...) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/core/modules/StarMgr.cpp:616
        m = {d = 0x5555587af0d0}
        catV = <optimized out>
        _container_ = {c = {<QListSpecialMethods<QVariant>> = {<No data fields>}, {p = {static shared_null = {ref = {atomic = {_q_value = {<std::__atomic_base<int>> = {static _S_alignment = 4, _M_i = -1}, static is_always_lock_free = true}}}, alloc = 0, begin = 0, 
                  end = 0, array = {0x0}}, d = 0x5555584e5300}, d = 0x5555584e5300}}, i = {i = 0x5555584e5338}, e = {i = 0x5555584e5358}, control = 1}
        cat_hip_sp_file_name = {d = 0x7ffff681e534 <QUtf8::convertToUnicode(char const*, int)+68>}
        cat_hip_cids_file_name = {d = 0x5555583d8b60}
#4  0x0000555555ab61b3 in StarMgr::init (this=this@entry=0x5555583d8220) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/core/modules/StarMgr.cpp:438
        conf = 0x555556c92740
        fic = {<QFileDevice> = {<QIODevice> = {<QObject> = {_vptr.QObject = 0x7ffff6a98518 <vtable for QFile+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, 
                    data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x5555583d8b60}, 
                static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                  superdata = {direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff6964720 <qt_meta_stringdata_QIODevice>, data = 0x7ffff6964620 <qt_meta_data_QIODevice>, 
                  static_metacall = 0x7ffff6717010 <QIODevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a98980 <QIODevice::staticMetaObject>}, 
                stringdata = 0x7ffff69645a0 <qt_meta_stringdata_QFileDevice>, data = 0x7ffff6964560 <qt_meta_data_QFileDevice>, static_metacall = 0x7ffff6711340 <QFileDevice::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, 
                extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff6a987e0 <QFileDevice::staticMetaObject>}, stringdata = 0x7ffff69643e0 <qt_meta_stringdata_QFile>, data = 0x7ffff69643a0 <qt_meta_data_QFile>, 
              static_metacall = 0x7ffff670dfe0 <QFile::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}
        app = <optimized out>
        displayGroup = {d = 0xffffffffffffff20}
#5  0x0000555555910412 in StelApp::init (this=this@entry=0x55555686aa20, conf=<optimized out>) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/core/StelApp.cpp:516
        cache = <optimized out>
        cachePath = {d = 0x555556d58420}
        hipsMgr = <optimized out>
        ssystem = <optimized out>
        nomenclature = <optimized out>
        hip_stars = 0x5555583d8220
        nebulas = <optimized out>
        milky_way = <optimized out>
        zodiacal_light = <optimized out>
        toasts = <optimized out>
        constellations = <optimized out>
        asterisms = <optimized out>
        landscape = <optimized out>
        gridLines = <optimized out>
        specialMarkers = <optimized out>
        meteors = <optimized out>
        skyLabels = <optimized out>
        skyMarkers = <optimized out>
        custObj = <optimized out>
        hlMgr = <optimized out>
#6  0x00005555556cc74a in StelMainView::init (this=0x7fffffffd4d0) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/StelMainView.cpp:815
        actionMgr = <optimized out>
        screen = <optimized out>
        screenGeom = {x1 = 1, y1 = 0, x2 = -159339548, y2 = 32767}
        size = {wd = -12320, ht = 32767}
        fullscreen = <optimized out>
        sgui = <optimized out>
#7  0x00005555556d2f32 in StelGLWidget::initializeGL (this=0x555556b135b0) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/StelMainView.cpp:134
        ctx = 0x555556b37290
#8  0x00007ffff781675f in QOpenGLWidget::event (this=0x555556b135b0, e=0x7fffffffd1a0) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qopenglwidget.cpp:1454
        d = 0x555556b19f40
#9  0x00007ffff77b296f in QApplicationPrivate::notify_helper (this=<optimized out>, receiver=0x555556b135b0, e=0x7fffffffd1a0)
    at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qapplication.cpp:3632
        consumed = false
        filtered = false
#10 0x00007ffff67c8a28 in QCoreApplication::notifyInternal2 (receiver=0x555556b135b0, event=0x7fffffffd1a0) at /mnt/portagetmp/portage/dev-qt/qtcore-5.15.2-r14/work/qtbase-f4ac0b55c37f2b594ffbe639db43dac365825c7c/src/corelib/kernel/qcoreapplication.cpp:1064
        selfRequired = true
        result = false
        cbdata = {0x555556b135b0, 0x7fffffffd1a0, 0x7fffffffd14f}
        d = <optimized out>
        threadData = 0x55555677b520
        scopeLevelCounter = {threadData = 0x55555677b520}
#11 0x00007ffff77f23e3 in QWidgetPrivate::show_helper (this=this@entry=0x555556b19f40) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qwidget.cpp:7814
        q = 0x555556b135b0
        isWindow = <optimized out>
        isEmbedded = false
        showEvent = {<QEvent> = {_vptr.QEvent = 0x7ffff7546c20 <vtable for QShowEvent+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff6975ec0 <qt_meta_stringdata_QEvent>, data = 0x7ffff6975900 <qt_meta_data_QEvent>, 
                static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}, d = 0x0, t = 17, posted = 0, spont = 0, m_accept = 1, reserved = 8191}, <No data fields>}
#12 0x00007ffff77f5563 in QWidgetPrivate::setVisible (this=0x555556b19f40, visible=<optimized out>) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qwidget.cpp:8112
        pw = <optimized out>
        wasResized = true
        initialWindowState = {i = 0}
        needUpdateGeometry = <optimized out>
        showToParentEvent = {_vptr.QEvent = 0x1, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff6975ec0 <qt_meta_stringdata_QEvent>, data = 0x7ffff6975900 <qt_meta_data_QEvent>, static_metacall = 0x0, relatedMetaObjects = 0x0, 
              extradata = 0x0}}, d = 0x7fffffffd278, t = 53888, posted = 1, spont = 1, m_accept = 1, reserved = 8191}
        q = 0x555556b135b0
#13 0x00007ffff77f2329 in QWidgetPrivate::showChildren (this=this@entry=0x5555569ccce0, spontaneous=spontaneous@entry=false) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qwidget.cpp:8184
        widget = <optimized out>
        i = 4
        childList = {<QListSpecialMethods<QObject*>> = {<No data fields>}, {p = {static shared_null = {ref = {atomic = {_q_value = {<std::__atomic_base<int>> = {static _S_alignment = 4, _M_i = -1}, static is_always_lock_free = true}}}, alloc = 0, begin = 0, end = 0, 
                array = {0x0}}, d = 0x5555569e9b60}, d = 0x5555569e9b60}}
#14 0x00007ffff77f23bf in QWidgetPrivate::show_helper (this=this@entry=0x5555569ccce0) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qwidget.cpp:7769
        q = 0x7fffffffd4d0
        isWindow = <optimized out>
        isEmbedded = <optimized out>
        showEvent = {<QEvent> = {_vptr.QEvent = 0x5555569ccce0, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff6975ec0 <qt_meta_stringdata_QEvent>, data = 0x7ffff6975900 <qt_meta_data_QEvent>, static_metacall = 0x0, 
                relatedMetaObjects = 0x0, extradata = 0x0}}, d = 0x7ffff77e243b <QWidgetPrivate::adjustedSize() const+203>, t = 0, posted = 0, spont = 0, m_accept = 0, reserved = 0}, <No data fields>}
#15 0x00007ffff77f5563 in QWidgetPrivate::setVisible (this=0x5555569ccce0, visible=<optimized out>) at /mnt/portagetmp/portage/dev-qt/qtwidgets-5.15.2-r12/work/qtbase-7c6c0030cf80ef7b9ace42996b0e0c3a72f76860/src/widgets/kernel/qwidget.cpp:8112
        pw = <optimized out>
        wasResized = false
        initialWindowState = {i = 0}
        needUpdateGeometry = <optimized out>
        showToParentEvent = {_vptr.QEvent = 0x1, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff6975ec0 <qt_meta_stringdata_QEvent>, data = 0x7ffff6975900 <qt_meta_data_QEvent>, static_metacall = 0x0, relatedMetaObjects = 0x0, 
              extradata = 0x0}}, d = 0x7fffffffd458, t = 10048, posted = 1, spont = 0, m_accept = 0, reserved = 2777}
        q = 0x7fffffffd4d0
#16 0x00005555556a5a5b in main (argc=<optimized out>, argv=<optimized out>) at /mnt/portagetmp/portage/sci-astronomy/stellarium-0.22.1/work/stellarium-0.22.1/src/main.cpp:389
        argStr = {d = 0x555556786960}
        argList = {<QList<QString>> = {<QListSpecialMethods<QString>> = {<No data fields>}, {p = {static shared_null = {ref = {atomic = {_q_value = {<std::__atomic_base<int>> = {static _S_alignment = 4, _M_i = -1}, static is_always_lock_free = true}}}, alloc = 0, 
                  begin = 0, end = 0, array = {0x0}}, d = 0x5555567868b0}, d = 0x5555567868b0}}, <No data fields>}
        app = {<QGuiApplication> = {<QCoreApplication> = {<QObject> = {_vptr.QObject = 0x7ffff7cdbf48 <vtable for QApplication+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, 
                    data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555556789360}, 
                static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                  superdata = {direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff6975440 <qt_meta_stringdata_QCoreApplication>, data = 0x7ffff6975320 <qt_meta_data_QCoreApplication>, 
                  static_metacall = 0x7ffff67ca470 <QCoreApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, static self = 0x7fffffffd490}, static staticMetaObject = {d = {superdata = {
                  direct = 0x7ffff6a9ba80 <QCoreApplication::staticMetaObject>}, stringdata = 0x7ffff74463e0 <qt_meta_stringdata_QGuiApplication>, data = 0x7ffff7446160 <qt_meta_data_QGuiApplication>, 
                static_metacall = 0x7ffff708b720 <QGuiApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {
                direct = 0x7ffff75461e0 <QGuiApplication::staticMetaObject>}, stringdata = 0x7ffff7b778a0 <qt_meta_stringdata_QApplication>, data = 0x7ffff7b77720 <qt_meta_data_QApplication>, 
              static_metacall = 0x7ffff77b66a0 <QApplication::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}
        envStelOpts = {d = 0x7fffec006ec0}
        versionLine = {d = 0x555556b4b700}
        copyrightLine = {d = 0x555556978710}
        maxLength = <optimized out>
        n = <optimized out>
        configName = {d = 0x55555700eb30}
        configFileFullPath = {d = 0x555556a1a8f0}
        confSettings = 0x555556c92740
        outputFile = {d = 0x555556a34700}
        fName = @0x7fffffffd468: {d = 0x555556d0a570}
        fileFont = {d = 0x555556d55160}
        baseFont = {d = 0x555556d5b480}
        tmpFont = {d = {d = 0x555556f38180}, resolve_mask = 3}
        trans = {<QTranslator> = {<QObject> = {_vptr.QObject = 0x5555566ed098 <vtable for CustomQTranslator+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, 
                  data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, d_ptr = {d = 0x555556979a00}, 
              static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {
                superdata = {direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff6989f60 <qt_meta_stringdata_QTranslator>, data = 0x7ffff6989f20 <qt_meta_data_QTranslator>, 
                static_metacall = 0x7ffff6801250 <QTranslator::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, <No data fields>}
        mainWin = {<QGraphicsView> = {<QAbstractScrollArea> = {<QFrame> = {<QWidget> = {<QObject> = {_vptr.QObject = 0x5555566ef260 <vtable for StelMainView+16>, static staticMetaObject = {d = {superdata = {direct = 0x0}, 
                        stringdata = 0x7ffff697a580 <qt_meta_stringdata_QObject>, data = 0x7ffff697a460 <qt_meta_data_QObject>, static_metacall = 0x7ffff67fa6f0 <QObject::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, 
                        extradata = 0x0}}, d_ptr = {d = 0x5555569ccce0}, static staticQtMetaObject = {d = {superdata = {direct = 0x0}, stringdata = 0x7ffff697d520 <qt_meta_stringdata_Qt>, data = 0x7ffff697a6a0 <qt_meta_data_Qt>, static_metacall = 0x0, 
                        relatedMetaObjects = 0x0, extradata = 0x0}}}, <QPaintDevice> = {_vptr.QPaintDevice = 0x5555566ef448 <vtable for StelMainView+504>, painters = 0, reserved = 0x0}, static staticMetaObject = {d = {superdata = {
                        direct = 0x7ffff6a931e0 <QObject::staticMetaObject>}, stringdata = 0x7ffff7b7b340 <qt_meta_stringdata_QWidget>, data = 0x7ffff7b7ac60 <qt_meta_data_QWidget>, 
                      static_metacall = 0x7ffff77f4160 <QWidget::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, data = 0x5555569cce00}, static staticMetaObject = {d = {superdata = {
                      direct = 0x7ffff7cdd360 <QWidget::staticMetaObject>}, stringdata = 0x7ffff7b907c0 <qt_meta_stringdata_QFrame>, data = 0x7ffff7b906c0 <qt_meta_data_QFrame>, 
                    static_metacall = 0x7ffff78a3b70 <QFrame::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff7ce04e0 <QFrame::staticMetaObject>}, 
                  stringdata = 0x7ffff7b90f20 <qt_meta_stringdata_QAbstractScrollArea>, data = 0x7ffff7b90e20 <qt_meta_data_QAbstractScrollArea>, static_metacall = 0x7ffff78a64c0 <QAbstractScrollArea::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, 
                  relatedMetaObjects = 0x0, extradata = 0x0}}}, static staticMetaObject = {d = {superdata = {direct = 0x7ffff7ce0ae0 <QAbstractScrollArea::staticMetaObject>}, stringdata = 0x7ffff7bc3660 <qt_meta_stringdata_QGraphicsView>, 
                data = 0x7ffff7bc3360 <qt_meta_data_QGraphicsView>, static_metacall = 0x7ffff7b246f0 <QGraphicsView::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x7ffff7d01fd0 <qt_meta_extradata_QGraphicsView>, extradata = 0x0}}}, 
          static staticMetaObject = {d = {superdata = {direct = 0x7ffff7d01fa0 <QGraphicsView::staticMetaObject>}, stringdata = 0x5555560652c0 <qt_meta_stringdata_StelMainView>, data = 0x555556064bc0 <qt_meta_data_StelMainView>, 
              static_metacall = 0x55555587bc10 <StelMainView::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)>, relatedMetaObjects = 0x0, extradata = 0x0}}, glInfo = {vendor = {d = 0x555556bbabd0}, renderer = {d = 0x555556ae25f0}, 
            mainContext = 0x555556b37290, functions = 0x7fffec006c20}, configuration = 0x555556c92740, rootItem = 0x555556978950, guiItem = 0x0, nightModeEffect = 0x770000007c, glWidget = 0x555556b135b0, stelScene = 0x5555569e84e0, gui = 0x555556c81280, 
          stelApp = 0x55555686aa20, updateQueued = false, flagInvertScreenShotColors = false, flagOverwriteScreenshots = false, flagUseCustomScreenshotSize = false, customScreenshotWidth = 1024, customScreenshotHeight = 768, screenshotDpi = 72, 
          customScreenshotMagnification = 1, screenShotPrefix = {d = 0x555556b13510}, screenShotFormat = {d = 0x5555569835a0}, screenShotDir = {d = 0x555556cdf150}, flagCursorTimeout = false, cursorTimeoutTimer = 0x555556978200, lastEventTimeSec = 0, minfps = 1, 
          maxfps = 10000, fpsTimer = 0x555556b13550}
```

</details>

@Kangie can you report the bug upstream?

**Comment by Kangie** [comment]
> Confirmed. Except that it also crashes after disabling stars.

So if built (and launched) with `USE=stars` and then rebuilt with `USE=-stars` we see the same behaviour?

**Comment by DarthGandalf** [comment]
yes

**Comment by alex-w** [comment]
See patch https://github.com/Stellarium/stellarium/commit/43d4dba4c85d3264244faad27cf1f60ddf92083c

**Comment by Kangie** [comment]
Amazingly fast fix on that one @alex-w!

File: sci-astronomy/stellarium/files/stellarium-0.22.1-fix-star-manager-segfault.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
+This patch is required to fix a segfault on startup in <=0.22.1 if star catalogue files are
+added or removed (USE=stars)
+
+Refer: https://github.com/Stellarium/stellarium/issues/2404
```
Comment (thesamesam) [inline sci-astronomy/stellarium/files/stellarium-0.22.1-fix-star-manager-segfault.patch:4]: nice, thanks for linking

**Pull request closed without merging**